### DISCUSSION TITLE
Auto3DSeg for Multi-Modal/Phase Input

### DISCUSSION THREAD

**Discussion by moonforsun** [Project-MONAI/tutorials#1071] [Q&A] [answered]
[Update on 01/03/2023]: The solution is using 4D data preparation. https://github.com/Project-MONAI/MONAI/issues/5592#issuecomment-1332702761

I am interested in how to get the auto3dseg work for the multi-phase/modal images as input. 
I can run the Task04_Hippocampus example but failed on my own multi-phase data.
My failure issue can be found here: https://github.com/Project-MONAI/MONAI/issues/5592.

**Comment by KumoLiu** [comment]
Hi @moonforsun, from the error message, It looks like you didn't prepare your data properly. Could you please prepare your data referring to this JSON file?
https://github.com/Project-MONAI/tutorials/blob/main/auto3dseg/tasks/msd/Task04_Hippocampus/msd_task04_hippocampus_folds.json
Hope it can help you, thanks!

**Comment by moonforsun** [comment, marked as answer]
[Update on 01/03/2023]: The answer is: Preparing the data in a 4D nifty can be the solution for this. https://github.com/Project-MONAI/MONAI/issues/5592#issuecomment-1332702761

@KumoLiu Thank you for your reply. Yes, the issue here is indeed due to the JSON. The reason is that I have multiple phase data for each case. 

Single phase/modality data that can be run in current Auto3DSeg version:
{
    "testing": [
        {
            "image": "imagesTs/hippocampus_267.nii.gz"
        },
        {
            "image": "imagesTs/hippocampus_379.nii.gz"
        },
...
}

Multiple phase/modality data that has the issue as in https://github.com/Project-MONAI/MONAI/issues/5592:
{
    "testing": [
        {
            "image": ["imagesTs/phase1.nii.gz", "imagesTs/phase2.nii.gz"]
        },
        {
            "image": ["imagesTs/phase1.nii.gz", "imagesTs/phase2.nii.gz"]
        },
...
}

Any suggestion of how should I prepare JSON for multiple phase data? Or any idea to support the multiple phase data as input for the Auto3DSeg?

**Reply by KumoLiu** [reply]
Hi @moonforsun, seems we didn't have an example about multiple modalities in auto3dseg. A simple way to do that is to concat modalities in the channel dim like this [tutorial](https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/brats_segmentation_3d.ipynb).
Thanks!

**Reply by moonforsun** [reply]
Preparing the data in a 4D nifty can be the solution for this. Thank you @KumoLiu for your nice reference to the tutorial.
